mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td

```diff
@@ -1409,9 +1417,10 @@ def Tosa_ReduceMaxOp : Tosa_InferTensorTypeOp<"reduce_max"> {
     /// Method used by InferTypeOpInterface.
     static bool isCompatibleReturnTypes(TypeRange l, TypeRange r);
 
-    /// Return the max of the two integer operands
-    static inline APInt calcOneElement(APInt leftOperand, APInt rightOperand) {
-      const llvm::APInt subtractRes = leftOperand - rightOperand;
+   /// Return the max of the two operands
+    template<typename T>
+    static std::optional<T> calcOneElement(T leftOperand, T rightOperand) {
```

Comment:
Does this correctly implement the behavior for NaNs as specified in `apply_max_s` on https://www.mlplatform.org/tosa/tosa_spec.html#_arithmetic_helpers?
Same question for reduce_min

Reply:
good point! I add the conditional check for NaNs